docs: add release process

pro-wh opened the pull request:
I found that Chrome wouldn't let me load an extension where the version is set to something like `1.0.0-rc1`, so we've come up with this thing where we'll use the patch number for release candidate builds leading up to a public release. This new document describes that.

[rendered](https://github.com/oasisprotocol/oasis-wallet-ext/blob/pro-wh/feature%2Fprocedures/docs/release-process.md)

kostko commented on docs/RELEASE-PROCESS.md:
```diff
@@ -0,0 +1,87 @@
+# Release process
+
+## Determine the new version
+Extension versions may not have suffixes like "-rc1."
```
```suggestion
Extension versions may not have suffixes like "-rc1".
```

pro-wh replied:
seeing as this is in the oasisprotocol org and many of its people are in europe, fine

kostko replied:
Oh interesting I didn't know this punctuation inside/outside quotes was a thing, it just looked wrong to me as it looked like the dot was part of the suffix :-)

kostko commented on docs/RELEASE-PROCESS.md:
```diff
@@ -0,0 +1,87 @@
+- public/manifest.json
+
+## Add a changelog section
+Here's the [changelog](changelog.md).
```
Should we rather use a top-level `CHANGELOG.md`?

pro-wh replied:
:shrug: that seems valid too. anyone have a preference? same for this doc, docs/ or top level? also all caps or lowercase? >_>

tjanez commented on docs/RELEASE-PROCESS.md:
```diff
@@ -0,0 +1,87 @@
+
+## Submit crx for review
+
+(Contact our web store account admins for this step.)
```
Since these instructions are already primarily intended for wallet maintainers, I guess we can just write the exact thing so we have everything in one place.
Something like:
Go to [Chrome Web Store Developer Dashboard](https://chrome.google.com/webstore/devconsole/), browse to the _Oasis Wallet_ item, select _Package_ in the side bar and choose _Upload new package_.

pro-wh replied:
gonna paste that in, thanks

tjanez commented on docs/RELEASE-PROCESS.md:
```diff
@@ -0,0 +1,87 @@
+
+Check the box to indicate that this is a pre-release.
+
+## Submit crx for review
```
crx -> Package

tjanez reviewed (commented):
Thanks for preparing this, @pro-wh, it will be really useful.

As for location and caps vs. lower case of this doc, we use `docs/release-process.md` in Oasis Core and some other Oasis projects :slightly_smiling_face:.

tjanez commented on docs/RELEASE-PROCESS.md:
```diff
@@ -0,0 +1,87 @@
+# Release process
```
(Applied to all headings in this doc)

pro-wh commented on docs/RELEASE-PROCESS.md:
````diff
@@ -0,0 +1,87 @@
+
+Our changelog is a file under version control, so all we do is put the link in the description:
+
+```md
````
ya

pro-wh commented:
new copy with everyone's suggestions, thanks

<details>
<summary>relevant diffs</summary>

```diff
diff --git a/docs/changelog.md b/CHANGELOG.md
similarity index 98%
rename from docs/changelog.md
rename to CHANGELOG.md
index 5211b90..c4c2e74 100644
--- a/docs/changelog.md
+++ b/CHANGELOG.md
@@ -1,4 +1,4 @@
-# Changelog
+# Change Log
 
 ## 1.0.0
 
diff --git a/README.md b/README.md
index c69b66f..72717d6 100644
--- a/README.md
+++ b/README.md
@@ -99,9 +99,18 @@ yarn test
 yarn buildProd
 ```
 
-## Preparing a Release
+## Preparing a Production Release Build
 
-Follow the applicable steps in the [release process doc](docs/RELEASE-PROCESS.md)
+```sh
+VERSION=<VERSION>
+rm -rf dist/
+yarn install --frozen-lockfile
+yarn buildProd
+zip -r oasis-wallet-$VERSION-$(git rev-parse --short HEAD).zip dist/
+```
+
+If you're actually making a new release, follow the applicable steps in the
+[release process doc](docs/release-process.md)
 
 ## LICENSE
 
diff --git a/docs/RELEASE-PROCESS.md b/docs/release-process.md
similarity index 60%
rename from docs/RELEASE-PROCESS.md
rename to docs/release-process.md
index 2983a3d..1a562e2 100644
--- a/docs/RELEASE-PROCESS.md
+++ b/docs/release-process.md
@@ -1,7 +1,7 @@
-# Release process
+# Release Process
 
-## Determine the new version
-Extension versions may not have suffixes like "-rc1."
+## Determine the New Version
+Extension versions may not have suffixes like "-rc1".
 We are incrementing patch versions to differentiate release candidate builds.
 
 For example, the 1.0 release will start with a build at version 1.0.0.
@@ -10,15 +10,15 @@ If changes are needed, we move to the next build at version 1.0.1.
 Version 1.0.1 is also a release candidate at first.
 If no changes are needed, 1.0.1 becomes the first public release of 1.0.
 
-## Set the new version throughout
+## Set the New Version Throughout
 You must update:
 
-- config.js
-- package.json
-- public/manifest.json
+- `config.js`
+- `package.json`
+- `public/manifest.json`
 
-## Add a changelog section
-Here's the [changelog](changelog.md).
+## Add a Change Log Section
+Here's the [Change Log](../CHANGELOG.md).
 If there is an "Unreleased changes" section, rename it to the new version.
 Do not include a "v" prefix in the heading.
 
@@ -34,16 +34,16 @@ Suggested sections:
 Items may include a link to GitHub issues if there is a relevant one, e.g. in bug fixes.
 We especially encourage linking externally reported issues.
 
-If you feel like it, move one changelog item to a section on top titled "Spotlight change."
+If you feel like it, move one Change Log item to a section on top titled "Spotlight change".
 
-## Propose the new version
-Open a pull request (PR) for the version changes and changelog update, ideally separate from
+## Propose the New Version
+Open a pull request (PR) for the version changes and Change Log update, ideally separate from
 any other changes.
 
 When this PR gets merged into `master`, the chosen version permanently associated with the merge
 commit.
 
-## Prepare an unsigned archive
+## Prepare an Unsigned Archive
 Clean the output directory, build, and package:
 
 ```sh
@@ -54,29 +54,34 @@ yarn buildProd
 zip -r oasis-wallet-$VERSION-$(git rev-parse --short HEAD).zip dist/
 ```
 
-## Add a pre-release on GitHub
+## Add a Pre-Release on GitHub
 Create a tag and create a
 [new release](https://github.com/oasisprotocol/oasis-wallet-ext/releases/new) on GitHub.
 
-TODO: standards on GitHub release title
+Enter "Oasis Wallet Release Candidate" for the title.
 
-Our changelog is a file under version control, so all we do is put the link in the description:
+Our Change Log is a file under version control, so all we do is put the link in the description:
 
 ```md
-_Note: This is an internal release candidate._
+See the [Change Log](CHANGELOG.md) for what's new in this release.
 
-See the [changelog](docs/changelog.md) for what's new in this release.
+_Note: This is an internal release candidate._
 ```
 
 Add the unsigned archive as an asset.
 
 Check the box to indicate that this is a pre-release.
 
-## Submit crx for review
+## Submit Package for Review
 
-(Contact our web store account admins for this step.)
+This step is for our web store account admins.
+Contact an admin if you don't have access.
+Go to [Chrome Web Store Developer Dashboard](https://chrome.google.com/webstore/devconsole/),
+browse to the _Oasis Wallet_ item,
+select _Package_ in the side bar and
+choose _Upload new package_.
 
-## Update GitHub pre-release
+## Update GitHub Pre-Release
 Change the description:
 
 ```diff
@@ -85,3 +90,5 @@ Change the description:
 ```
 
 Uncheck the pre-release box.
+
+Upload the artifact without the Git commit hash.
```

</details>

tjanez commented on docs/RELEASE-PROCESS.md:
```diff
@@ -0,0 +1,87 @@
+Create a tag and create a
+[new release](https://github.com/oasisprotocol/oasis-wallet-ext/releases/new) on GitHub.
+
+TODO: standards on GitHub release title
```
Uh, I wrote `"Oasis Wallet <VERSION> Release Candidate"`, but GitHub Markdown renderer just ate the `<VERSION>` part...

tjanez replied:
@pro-wh, I was basically codifying what you did with "Oasis Wallet 1.0.0 Release Candidate".

tjanez replied:
I would include the version in the title. @pro-wh, what do you think?

pro-wh replied:
oops yeah

tjanez commented on docs/release-process.md:
```diff
@@ -0,0 +1,92 @@
+
+## Submit Package for Review
+
+This step is for our web store account admins.
```
```suggestion
_This step is for our web store account admins._
```

tjanez commented on docs/release-process.md:
```diff
@@ -0,0 +1,92 @@
+## Submit Package for Review
+
+This step is for our web store account admins.
+Contact an admin if you don't have access.
```
```suggestion
_Contact an admin if you don't have access._
```

tjanez commented on docs/release-process.md:
```diff
@@ -0,0 +1,92 @@
+
+This step is for our web store account admins.
+Contact an admin if you don't have access.
+Go to [Chrome Web Store Developer Dashboard](https://chrome.google.com/webstore/devconsole/),
```
Add a new line before this line.

tjanez replied:
I.e. after "Contact an admin if you don't have access."

pro-wh replied:
ok

tjanez commented on docs/release-process.md:
```diff
@@ -0,0 +1,95 @@
+rm -rf dist/
+yarn install --frozen-lockfile
+yarn buildProd
+zip -r oasis-wallet-$VERSION-$(git rev-parse --short HEAD).zip dist/
```
```suggestion
zip -r oasis-wallet-$VERSION.zip dist/
```

pro-wh replied:
ok I'll remove that last line about re-uploading without the commit hash too

lukaw3d replied:
Heads up: For Firefox you'll need `(cd dist && zip --recurse-paths ../oasis-wallet-$VERSION.zip ./)`

And it should work for Chrome too

pro-wh replied:
filed #196, thanks

pro-wh commented:
fixed missing version from release title

added a step to remove "Release Candidate" from gh rel

add a note not to add git commit hash to the unsigned archive filename

tjanez reviewed (approved):
Great work, @pro-wh!
